zapcore/encoder.go

```diff
@@ -112,21 +112,43 @@ func EpochNanosTimeEncoder(t time.Time, enc PrimitiveArrayEncoder) {
 	enc.AppendInt64(t.UnixNano())
 }
 
+type appendTimeEncoder interface {
+	AppendTimeLayout(time.Time, string)
+}
+
+func encodeTimeLayout(t time.Time, layout string, enc PrimitiveArrayEncoder) {
+	if enc, ok := enc.(appendTimeEncoder); ok {
```

Comment:
fwiw: it's possible to just use an anonymous `duck, ok := val.(interface { quack() })` in cases like this where aliasing the interfaces serves no other purpose

Reply:
Yep, although the line looks kind of long and complicated.

However, I moved the interface into the method, since it's not needed outside of this method.